pkg/fleetautoscalers/controller.go

```diff
@@ -292,6 +292,11 @@ func (c *Controller) syncFleetAutoscaler(ctx context.Context, key string) error
 		return err
 	}
 
+	// Don't do anything, the fleet is marked for deletion
+	if !fleet.DeletionTimestamp.IsZero() {
```

Comment:
Are there any other cases where this timestamp might be set to be non-zero?

According to https://kubernetes.io/docs/concepts/architecture/garbage-collection/#foreground-deletion if the object is being deleted using foreground deletion it will also be apparent because the `metadata.finalizers` field will be set to `foregroundDeletion` in addition to having a deletion timestamp set. I'm curious if there is any reason to check the finalizers here in addition to the deletion timestamp.

Reply:
🤔 I'm _thinking_ that we don't need to check it, just because, regardless of if it's a foreground or background deletion, we shouldn't have activity in Fleets or related - it's a waste of resources.

Reply:
I'm not sure it's necessary to check for the finalizer specifically, we are mostly interested in knowing that the resource has been marked for deletion so as to not take any action that will end up in the recreation of gameservers unnecessarily.
The other case when DeletionTimestamp is set is for Orphan Cascading, which for FleetAutoScalers doesn't mean anything special as they are not owners of any resources and can be considered as a regular deletion, and for GameserverSets would mean it shouldn't add new game servers and the existing ones shouldn't be removed. For the latter, the current PR is setting `numServersToAdd` to 0 so it is consistent with that behaviour.